1.3 Spring Boot Security commit. Spring Security implementation.

kopervas opened the pull request.

kopervas committed:
1.3 Spring Boot Security commit. Spring Security implementation.

Elena-Bruyako commented on src/main/java/com/bookstore/config/SecurityConfig.java:
```diff
@@ -0,0 +1,46 @@
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
```
```suggestion
@EnableMethodSecurity
@Configuration
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/com/bookstore/config/SecurityConfig.java:
```diff
@@ -0,0 +1,46 @@
+                        .logoutUrl("/logout")
+                );
+
+        return http.build();
```
```suggestion

        return http
                .cors(AbstractHttpConfigurer::disable)
                .csrf(AbstractHttpConfigurer::disable)
                .authorizeHttpRequests(
                        auth -> auth
                                .requestMatchers(
                                        antMatcher("/auth/**"),
                                        antMatcher("/swagger-ui/**"),
                                        antMatcher("/v3/api-docs/**"))
                                .permitAll()
                                .anyRequest()
                                .authenticated()
                )
                .sessionManagement(
                        s -> s.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
                .userDetailsService(service)
                .build();
```

Elena-Bruyako commented on src/main/java/com/bookstore/config/SecurityConfig.java:
```diff
@@ -0,0 +1,46 @@
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+public class SecurityConfig {
```
```suggestion
public class SecurityConfig {

    private final UserDetailsService service;
```

Elena-Bruyako commented on src/main/java/com/bookstore/controller/AuthenticationController.java:
```diff
@@ -16,7 +16,7 @@
 
 @RestController
-@RequestMapping("/auth")
+@RequestMapping("/api/auth")
```
add `server.servlet.context-path=/api` to `application.properties`
```suggestion
@RequestMapping("/auth")
```

Elena-Bruyako commented on src/main/java/com/bookstore/controller/BookController.java:
```diff
@@ -14,12 +14,11 @@
 @Tag(name = "Product management", description = "Endpoints for managing products")
 @RequiredArgsConstructor
-@RestController
+//@HomeController
```
```suggestion
@RestController
```

Elena-Bruyako commented on src/main/java/com/bookstore/controller/HomeController.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HomeController {
```
remove

Elena-Bruyako commented on src/main/java/com/bookstore/model/Role.java:
```diff
@@ -0,0 +1,33 @@
+import jakarta.persistence.Table;
+import org.springframework.security.core.GrantedAuthority;
+
+@Entity
```
```suggestion
@Entity
@Getter
@Setter
```

Elena-Bruyako commented on src/main/java/com/bookstore/model/Role.java:
```diff
@@ -0,0 +1,33 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "name", nullable = false, unique = true)
```
```suggestion
    @Column(nullable = false, unique = true)
```

Elena-Bruyako commented on src/main/java/com/bookstore/model/User.java:
```diff
@@ -30,8 +39,47 @@ public class User {
+            name = "user_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
```
```suggestion
    @ManyToMany
    @JoinTable(
            name = "users_roles",
            joinColumns = @JoinColumn(name = "user_id"),
            inverseJoinColumns = @JoinColumn(name = "role_id")
    )
```

Elena-Bruyako commented on src/main/java/com/bookstore/model/User.java:
```diff
@@ -30,8 +39,47 @@ public class User {
     private String shippingAddress;
 
-    @Column(nullable = false)
+    @Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT(1)")
```
```suggestion
    @Column(nullable = false, columnDefinition = "TINYINT(1)")
```

Elena-Bruyako commented on src/main/java/com/bookstore/model/User.java:
```diff
@@ -30,8 +39,47 @@ public class User {
+
+    @Override
+    public boolean isEnabled() {
+        return UserDetails.super.isEnabled();
```
```suggestion
        return !isDeleted;
```

Elena-Bruyako commented on src/main/java/com/bookstore/repository/UserRepository.java:
```diff
@@ -3,10 +3,13 @@
 
     boolean existsByEmail(String email);
 
+    Optional<User> findByEmail(String email);
```
```suggestion
    @EntityGraph(attributePaths = "roles")
    Optional<User> findByEmail(String email);
```

Elena-Bruyako commented on src/main/java/com/bookstore/service/impl/UserServiceImpl.java:
```diff
@@ -26,10 +32,14 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
+        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+
+        Role roleUser = roleRepository.findByRole(Role.RoleName.USER)
+                .orElseThrow(() -> new RegistrationException("Default role USER not found"));
```
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Default role " + Role.RoleName.USER + " not found"));
```

Elena-Bruyako commented on src/main/java/com/bookstore/service/impl/UserServiceImpl.java:
```diff
@@ -4,18 +4,24 @@
 
+import java.util.Set;
+
 @Service
```
```suggestion
@Service
@Transactional
```

Elena-Bruyako commented on src/main/java/com/bookstore/service/impl/UserServiceImpl.java:
```diff
@@ -4,18 +4,24 @@
-import org.springframework.data.domain.Pageable;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
```
checkstyle doesn't work, fix please

Elena-Bruyako commented on src/main/resources/db/changelog/changes/create-users-roles-table.yaml:
```diff
@@ -0,0 +1,37 @@
+      author: serg.kopervas
+      changes:
+        - createTable:
+            tableName: user_roles
```
```suggestion
            tableName: users_roles
```

Elena-Bruyako reviewed (changes requested):
fix CI/CD
fix checkstyle
see comments

kopervas committed:
1.4 Spring Boot Security commit. Errors correction.

sarakhmen commented on src/main/java/com/bookstore/model/Role.java:
```diff
@@ -8,15 +8,20 @@
     private Long id;
 
+    //@Column(name = "name", nullable = false, unique = true)
     @Column(name = "name", nullable = false, unique = true)
```
still not fixed. Also please remove commented line

sarakhmen commented on src/main/java/com/bookstore/model/User.java:
```diff
@@ -39,17 +39,18 @@ public class User implements UserDetails {
     private String shippingAddress;
 
-    @Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT(1)")
+    //@Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT(1)")
```
remove commented line, fix same issue in all places

sarakhmen commented on src/main/resources/db/changelog/changes/set-default-role-to-users.yaml:
```diff
@@ -3,27 +3,26 @@ databaseChangeLog:
               WHERE u.email = 'admin@example.com'
               AND NOT EXISTS (
-                  SELECT 1 FROM user_roles ur
+                  SELECT 1 FROM users_roles ur
```
why do you have 2 liquibase changeset files for assigning user roles?

sarakhmen reviewed (changes requested):
Good job! But let's improve your solution one more time :)

kopervas committed:
1.5 Spring Boot Security commit. Errors correction by Artur Sarakhman:
1) corrected column name for Role.role from "name" to "role";
2) removed commented line in all places;
3) unnecessary liquibase changeset file "assign-roles-to-users.yaml" was deleted.
